### PR TITLE
Promtail: Add text encoding conversion to file targets

### LINKED ISSUE #1119: Configure encoding of scraped logs
As Windows is... special... in the Server2019 edition (at least) the log files are saved with the `UTF-16` encoding, this means that trying to scrape them as-is will result in a mangled output in Grafana.

If it was possible to specify through configuration the encoding of the scraped logs (and then parsing the files accordingly) this problem above could be simply solved.

For the time being I'm re-saving the files with `UTF-8` encoding so that they can be displayed correctly.

### PR DESCRIPTION
**Pull request by vyzigold**
**What this PR does / why we need it**:
This PR adds support for text encoding conversion for file targets. To use this, just add `encoding: <encoding_name>` into the scrapeconfig like this:
```
scrape_configs:
- job_name: system_convert
  encoding: utf-16
  static_configs:
  - targets:
      - localhost
    labels:
      job: varlogs
      __path__: /var/utf16.log
```
The encoding name should be one of the names described by IANA [here](https://www.iana.org/assignments/character-sets/character-sets.xhtml).

**Which issue(s) this PR fixes**:
Fixes #1119 

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: pkg/promtail/targets/file/tailer.go [hunk 1]
```diff
@@ -120,9 +145,20 @@ func (t *tailer) run() {
 				continue
 			}
 
+			var text string
+			if t.encoding != "" {
+				var err error
+				text, err = t.convertToUTF8(line.Text)
+				if err != nil {
+					level.Error(t.logger).Log("msg", "failed to convert encoding", "error", err)
```
Comment (owen-d) [inline pkg/promtail/targets/file/tailer.go:153]: I wonder if it's worth adding a metric here for encoding attempts.

**Review by owen-d — COMMENTED** [review]
This is an interesting one, but I'm curious how useful ingesting different encodings is without the ability to change encoding again on the read path?

This PR does stand on it's own though and it could be an interesting first step towards supporting it on the read path (for now consumers would be responsible for encoding changes during reads).

/cc @cyriltovena @slim-bean

**Comment by snv** [comment]
any possibility of merging this PR ?

**Comment by slim-bean** [comment]
@vyzigold do you have any interest/time to reopen this? If not I'd like to reopen it probably by copying your work and giving you a co-authored-by in the commits. But this was useful work and I'm sorry we didn't get it merged before but would like to fix that!

**Comment by yosiasz** [comment]
wonder why this was closed?

**Pull request closed without merging**